src/test/java/org/sonar/plugins/scm/jazzrtc/JazzRtcBlameCommandTest.java

```diff
@@ -118,6 +120,38 @@ public Integer answer(InvocationOnMock invocation) throws Throwable {
         new BlameLine().date(DateUtils.parseDateTime("2014-12-09T09:14:00+0000")).revision("1000").author("Julien HENRY")));
   }
 
+  /**
+   * Assuming that CommandExecutor times out, tests if we break execution immediately with a IllegalStateException and that the timeout in the config is
+   * passed to the executor.
+   */
+  @Test
+  public void testCommandTimeout() throws IOException {
+    long testTimeout = 100l;
+
+    JazzRtcConfiguration mockedConfig = mock(JazzRtcConfiguration.class);
+    when(mockedConfig.commandTimeout()).thenReturn(testTimeout);
+
+    DefaultInputFile inputFile = createTestFile("src/foo.xoo", 3);
+
+    when(commandExecutor.execute(any(Command.class), any(StreamConsumer.class), any(StreamConsumer.class), anyLong())).thenAnswer(new Answer<Integer>() {
```

Comment:
If I remember correctly there is a when(xxx).doThrow(yyy) construction that would avoid the complex Answer construct